### PR TITLE
try noarch: python

### PR DESCRIPTION
**Pull request by looooo**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by looooo** [comment]
Is there a difference between ninja for pypy and ninja for python?

**Comment by looooo** [comment]
ok so windows need python to compile. But still I would prefer to make python an optional runtime-dependency. Is there any way to add python as a host-dependency but not add it to the run-dependencies? noarch:python?

**Comment by frol** [comment]
@jakirkham 

> Is there any way to add python as a host-dependency but not add it to the run-dependencies? noarch:python?

**Comment by looooo** [comment]
@conda-forge/core Please help with this recipe. We need to have python in host environment for the build (windows) But run-dependency on python should be optional. How to solve this?

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by wolfv** [comment]
I think we can use the CMake build to remove the dependency on Python.

https://github.com/ninja-build/ninja/blob/master/CMakeLists.txt

do you have cycles to try this?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,16 +10,16 @@ source:
...
-  number: 1
+  number: 2
   skip: true  # [win and vc!=14]
   ignore_run_exports:
     - vc
 
 requirements:
   build:
     - {{ compiler("cxx") }}
+    - cmake
```
Comment (wolfv) [inline recipe/meta.yaml:21]: probably need to add `make` here

Comment (wolfv) [reply recipe/meta.yaml:21]: for unix

**Comment by isuruf** [comment]
You need to move the `python` dependency to `build` instead of `host`.

**Comment by wolfv** [comment]
@isuruf you tried that above, no?

**Comment by isuruf** [comment]
I don't remember. Logs are gone.

**Comment by wolfv** [comment]
https://github.com/conda-forge/ninja-feedstock/pull/19/commits/7086e78dee5af9570d23d912f9e6fd886342a04e

**Comment by isuruf** [comment]
What was the failure with that commit though?

**Comment by wolfv** [comment]
windows didn't work from what I have understood by reading the comments here.

**Comment by looooo** [comment]
sry, I cannot remember.

**Comment by looooo** [comment]
I tired it again and this is the error:
`'D:\bld\ninja_1598436865888\_h_env\python.exe' is not recognized as an internal or external command`

**Comment by isuruf** [comment]
`%PYTHON% configure.py --bootstrap` should be changed to `python configure.py --bootstrap`

**Comment by looooo** [comment]
fixed in https://github.com/conda-forge/ninja-feedstock/pull/22

**Pull request closed without merging**